### ISSUE TITLE
No Stop() method

### ISSUE THREAD

**Issue by dd8** [MicrosoftEdge/WebViewFeedback#28]
Windows.UI.Xaml.Controls.WebView has Stop() method
MFC CHtmlView has Stop() method
MsHtml IWebBrowser2 has Stop() method
macOS WkWebView has stopLoading() method

There is a put_Cancel method on IWebView2NavigationStartingEventArgs, but communicating a stop command from the UI to the navigation delegate is a lot more complex than just calling webView->Stop()

Given modern browser UIs merge the Stop and Refresh buttons I'd expect to be able to write something as simple as this:

```
OnClickedStopOrRefresh()
{
 if ( IsLoading() )
  webView->Stop();
 else
  webView->Refresh();
}
```

**Comment by liminzhu** [comment]
Thanks for the feedback! Sounds reasonable to me that you'd want a cancel/stop functionality outside of the navigationStarting event. We will look into it.

If you don't mind me asking, what are you planning to build with WebView?

**Comment by dd8** [comment]
We have a couple of desktop products that use WebViews for part of the UI, and for capturing page thumbnails. On Windows we've used MsHtml since IE 3.0 was released, and on Mac we use the WebKit Framework. We're planning to migrate the Windows products to the Edge/Chromium WebView Windows.

The products are here:
https://www.powermapper.com/products/sortsite/
https://www.powermapper.com/products/mapper/ 

You can see how the WebView is used here. This is a Mac screenshot, but our Windows UI is being updated to look like this:
<img width="1250" alt="Screen Shot 2019-06-28 at 20 51 42" src="https://user-images.githubusercontent.com/16352274/60368157-ff3a0b80-99e7-11e9-9997-c5f650d13f91.png">

**Comment by liminzhu** [comment]
Thanks, this is really cool to see!

**Comment by dd8** [comment]
We have a working prototype using WebView2. The dev preview was simple to integrate (took less than a day to get a working UI) and seems very stable so far. Still a few things to do, so we'll probably find a few more feature gaps.

**Comment by liminzhu** [comment]
Hey @dd8 , as requested we've made a new `Stop` API in the latest [release](https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview2/releasenotes#08230). Let us know if this work for ya!

**Comment by verelpode** [comment]
Thanks for creating the `Stop` method -- that was certainly a useful addition.  However, what about stopping script execution?  Does or should the [Stop](https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview2/reference/iwebview2webview3#stop) method stop the running of all scripts in the webpage?  I'd like to suggest that the documentation for `Stop` be updated to explain how it does or doesn't affect scripts.  Currently it says only:
> "Stop all navigations and pending resource fetches."

**If** the final decision is that `Stop` shouldn't stop scripts, then I suggest:
1. Rename `Stop` to `StopLoading`; and
2. Create a method named `StopExecutingScripts` or `StopScriptExecution`.  

Here are a couple of examples of the usefulness of stopping scripts:
* A "Stop" button similar to a "Stop" button in a web browser could stop downloading, loading, fetches, and all scripts in the webpage.
* A `StopScriptExecution` method would also be useful in conjunction with the existing [ExecuteScript](https://docs.microsoft.com/en-us/microsoft-edge/hosting/webview2/reference/iwebview2webview#executescript) method.

Also, why does `Stop` appear in **both** of `IWebView2WebView2` and `IWebView2WebView3`?

Will `IWebView2WebView2` and `IWebView2WebView3` be removed in the first release version of WebView2 and their members merged into `IWebView2WebView`?  I hope so.  Prior to the first release, while WebView2 is still in beta, I think it's acceptable for an interface (`IWebView2WebView`) to be changed in a way that breaks compatibility with the previous beta version.  There is no need to retain backwards-compatibility baggage before the first release version.

**Comment by david-risney** [comment]
Stop was added for folks creating browser like scenarios and so that guides how Stop behaves currently. I'll look into how it interacts with script and get that updated (#50 ). We can look into improvements to Stop related functionality in the future.

Stop appears in both interfaces because we want to have the version suffixed interfaces derive from their previous version so we only need to have a pointer to the latest version suffixed interface. IWebView2WebView2 only had Stop on it, so rather than making a breaking change to the interfaces we just added Stop to IWebView2WebView3. We plan to make a breaking change for v1 where we merge all the version suffixed interfaces back down to just one interface. So yes the duplicate Stop is awkward now, but will be addressed in a future breaking change and for sure before we're out of developer preview.
